libs/RobotKit/include/RobotController.h

```diff
@@ -312,10 +329,13 @@ class RobotController : public interface::RobotController
 
 	auto isBleConnected() -> bool final { return _is_ble_connected; }
 
+	void connectToBle() final { _is_ble_connected = true; }
```

Comment:
je suis pas très fan de cette fonction qui n'a d'usage que pour les tests, pour plusieurs raisons:

1. le nom n'est pas adapté parce qu'elle ne fait pas ce qu'elle dit, qu'elle va faire, elle ne connecte pas au BLE.

le nom suivant serait plus adapté:

```suggestion
	void setBLEConnectionStatusToTrue() final { _is_ble_connected = true; }
```

2. la connection au BLE est orthogonale au RC. du coup ça fait bizarre d'avoir une dépendance si forte avec cette fonction qui donne l'impression que le RC peut décider ou non de connecter en BLE. Connecter à quoi d'ailleurs puisque c'est de toute manière pas le robot qui initie le truc.

dans le cas présent, et comme la connection est orthogonal, pourquoi dans les tests ne pas passer avant par un event "ble connection" afin de lancer `startConnectionBehavior` et d'être dans cet état de connection.

on peut aussi passer par `isBleConnected` + la sm directement pour avoir le statut avec  avec `sm.is(lksm::state::connected)`

```cpp
auto isBleConnected() -> bool final { return state_machine.is(lksm::state::working); }
```

ça serait même le plus smart maintenant qu'on a la machine orthogonale puisque comme pour `charging` on peut verifier qu'on est bien toujours `connected` avec un guard (j'imagine que BLEKit ou une de ses dépendances peut nous dire si on est connecté ou non)